data_steward/participant_validation/validate_email.py

```diff
@@ -0,0 +1,121 @@
+# Python imports
+import logging
+import argparse
+
+# Project imports
+from utils import bq, pipeline_logging, auth
+from tools.create_tier import SCOPES
+from common import JINJA_ENV, PS_API_VALUES, DRC_OPS
+
+LOGGER = logging.getLogger(__name__)
+
+EHR_OPS = 'ehr_ops'
+
+CREATE_EMAIL_COMPARISON_FUNCTION = JINJA_ENV.from_string("""
+    CREATE FUNCTION IF NOT EXISTS `{{project_id}}.{{drc_dataset_id}}.CompareEmail`(rdr_email string, ehr_email string)
+    RETURNS string
+    AS ((
+        WITH normalized_rdr_email AS (
+            SELECT LOWER(TRIM(rdr_email)) AS rdr_email
+        )
+        , normalized_ehr_email AS (
+            SELECT LOWER(TRIM(ehr_email)) AS ehr_email
+        )
+        SELECT
+            CASE 
+                WHEN normalized_rdr_email.rdr_email = normalized_ehr_email.ehr_email
+                    AND REGEXP_CONTAINS(normalized_rdr_email.rdr_email, '@') THEN 'match'
+                WHEN normalized_rdr_email.rdr_email IS NOT NULL AND normalized_ehr_email.ehr_email IS NOT NULL THEN 'no_match'
+                WHEN normalized_rdr_email.rdr_email IS NULL THEN 'missing_rdr'
+                ELSE 'missing_ehr'
+            END AS email
+        FROM normalized_rdr_email, normalized_ehr_email 
+
+    ));
+""")
+
+MATCH_EMAIL_QUERY = JINJA_ENV.from_string("""
+    UPDATE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}` upd
+    SET upd.email = `{{project_id}}.{{drc_dataset_id}}.CompareEmail`(ps.email, ehr.email),
+        upd.algorithm = 'yes'
+    FROM `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}` id_match
+    LEFT JOIN `{{project_id}}.{{drc_dataset_id}}.{{ps_api_table_id}}` ps
+        ON ps.person_id = id_match.person_id
+    LEFT JOIN `{{project_id}}.{{ehr_ops_dataset_id}}.{{hpo_pii_email_table_id}}` ehr
+        ON ehr.person_id = id_match.person_id
+    WHERE upd.person_id = id_match.person_id
```

Comment:
I think this might need to be a [MERGE](https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax#merge_statement) since new participants can be added by ps or the site. I'm trying to test out some things to see how it works for this instance.